packages/core/src/modules/basic-messages/BasicMessagesModule.ts

```diff
@@ -1,5 +1,9 @@
 import type { DependencyManager, Module } from '../../plugins'
+import type { FeatureRegistry } from '../discover-features'
 
+import { Protocol } from '../discover-features'
```

Comment:
Same here. I think the protocol / feature registry should be part of core so not all modules depend on the discover features module. This way the discover features module just exposes it as a protocol, but the internal registration is in core. What do you think?